### PR TITLE
Use 1ES Pools

### PR DESCRIPTION
**Pull request by jonfortescue**
- [x] I've read the guidelines for [contributing](CONTRIBUTING.md) and seen the [walkthrough](https://youtu.be/9OMxy1wal1s?t=1869)
- [ ] I've posted a comment on an issue with a detailed description of how I am planning to contribute and got approval from a member of the team
- [ ] The code builds and tests pass locally (also verified by our automated build checks)
- [x] Commit messages follow this format:
```
        Summary of the changes
        - Detail 1
        - Detail 2

        Fixes #bugnumber
```
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Code follows the same patterns and style as existing code in this repo

Same as https://github.com/dotnet/efcore/pull/25807.

### CODE REVIEW

File: azure-pipelines-richnav.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ stages:
             enablePublishTestResults: true
             pool:
               name: NetCorePublic-Pool
```
Comment (smitpatel) [inline azure-pipelines-richnav.yml:21]: Looking at https://github.com/dotnet/efcore/pull/25808/files
Should this name also be updated?

Comment (jonfortescue) [reply azure-pipelines-richnav.yml:21]: it should! I just missed that one it looks like; thank you for the catch

**Review by ulisesh — APPROVED** [review]
LGTM

**Comment by wtgodbe** [comment]
yeah, don't reset rc1 for this, target it against 'release/6.0 instead

**Comment by smitpatel** [comment]
This change is already in release/6.0 branch

**Comment by jonfortescue** [comment]
If we don't need it in this branch, I'll just close this PR since it's already in release/6.0.

**Pull request closed without merging**